spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,396 @@
+          def saved_change_to_title_date?; end
+
+          sig { returns(T.nilable(Date)) }
+          def title_date; end
+
+          sig { params(title_date: T.nilable(Date)).returns(T.nilable(Date)) }
+          def title_date=(title_date); end
+
+          sig { returns(T::Boolean) }
+          def title_date?; end
+
+          sig { returns(T.nilable(Date)) }
+          def title_date_before_last_save; end
+
+          sig { returns(T.nilable([T.nilable(Date), T.nilable(Date)])) }
+          def title_date_change; end
+
+          sig { returns(T::Boolean) }
+          def title_date_changed?; end
+
+          sig { returns(T.nilable(Date)) }
+          def title_date_was; end
+        end
+        RUBY
+
+      expect(rbi_for(content)).to(eq(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with datetime ") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.datetime(:review_date)
+          end
+
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(DateTime)) }
+          def review_date; end
+
+          sig { params(review_date: T.nilable(DateTime)).returns(T.nilable(DateTime)) }
+          def review_date=(review_date); end
+
+        RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with time ") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.time(:review_time)
+          end
+
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(Time)) }
+          def review_time; end
+
+          sig { params(review_time: T.nilable(Time)).returns(T.nilable(Time)) }
+          def review_time=(review_time); end
+
+        RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with decimal type ") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.decimal(:rate)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(BigDecimal)) }
+          def rate; end
+
+          sig { params(rate: T.nilable(BigDecimal)).returns(T.nilable(BigDecimal)) }
+          def rate=(rate); end
+        RUBY
+      expect(rbi_for(content)).to(include(expected))
+    end
+
+    it("generate RBI for simple TypedStore classes with any type ") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.any(:kind)
+          end
+        end
+      RUBY
+
+      expected = <<~RUBY
+        # typed: strong
+        class Post
+          sig { returns(T.nilable(T.untyped)) }
```

Comment:
This is probably something we should fix. `T.nilable(T.untyped)` is meaningless. We should not be wrapping a `T.untyped` in a `T.nilable` at all.